File: app/models/semester.rb
class Semester < BaseDocument

  ## Currently we define semester as a:
  ## 1. Season (Fall or Spring)
  ## 2. Year (YYYY)
  ##    ex. Fall 2009, Spring 2011, Spring 2014

  ##Fields
  field :season, :type => String
  field :year, :type => String

  ## Relationships
  belongs_to :teacher
  has_many :courses

end

File: app/assets/javascripts/models/semester.js
Sis.Semester = DS.Model.extend({
  season: DS.attr('string'),
  year: DS.attr('string'),
  courses: DS.hasMany('course'),
  teacher: DS.belongsTo('teacher'),

  title: function(key, value) {
    if (this.get('season') && this.get('year')) {
      return this.get('season') + " " + this.get('year');
    } else {
      return '';
    }
  }.property('season', 'year'),
});